scipy/linalg/tests/test_lapack.py

```diff
@@ -3506,3 +3505,43 @@ def test_lantr(norm, uplo, m, n, diag, dtype):
     ref = lange(norm, A)
 
     assert_allclose(res, ref, rtol=2e-6)
+
+
+@pytest.mark.parametrize('dtype', DTYPES)
+@pytest.mark.parametrize('norm', ['1', 'I', 'O'])
+def test_gbcon(dtype, norm):
+    rng = np.random.default_rng(17273783424)
+
+    # A is of shape n x n with ku/kl super/sub-diagonals
+    n, ku, kl = 10, 2, 2
+    A = rng.random((n, n)) + rng.random((n, n))*1j
+    # make the condition numbers more interesting
+    offset = rng.permuted(np.logspace(0, rng.integers(0, 10), n))
+    A += offset
+    if np.issubdtype(dtype, np.floating):
+        A = A.real
+    A = A.astype(dtype)
+    A[np.triu_indices(n, ku + 1)] = 0
+    A[np.tril_indices(n, -kl - 1)] = 0
+
+    # construct banded form
+    LDAB = 2*kl + ku + 1
```

Comment:
I see in `solve_banded` we also provide the LAPACK routine with a matrix with `2 * kl  +  ku  +  1` rows. IIUC, the first `kl` rows are all zero. I imagine this is used as work space by the routine, but do you know if it is necessary?

(Just got information... I suppose you have to do whatever it takes to use the `gbtrf`.)